tollway/__main__.py

```diff
@@ -65,27 +72,36 @@ def main(
         vehicle = create_vehicle(fake=fake)
         message = create_message(vehicle=vehicle, tollway=tollway)
 
-        # DATE VARIATION
-        if date_variation and event_count % DATE_VARIATION_RATE == 0:
-            message["timestamp"] = get_date_variation(timestamp=message["timestamp"])
-
         # LATE EVENTS
-        if include_late:
-            events_log["past_events_timestamps"].append(message["timestamp"])
-            if len(events_log["past_events_timestamps"]) == INCLUDE_LATE_RATE:
-                events_log = process_late_event(
-                    events_log=events_log,
-                    fake=fake,
-                    tollways=tollways,
-                    publisher=publisher,
-                    topic_path=topic_path,
-                )
-                include_late_processed = True
+        if include_late_seconds:
+            events_log["late_events"]["seconds"].append(message["timestamp"])
+
+        if include_late_minutes:
+            events_log["late_events"]["minutes"].append(message["timestamp"])
+
+        if include_late_hours:
+            events_log["late_events"]["hours"].append(message["timestamp"])
+
+        if include_late_days:
+            events_log["late_events"]["days"].append(message["timestamp"])
+
+        if include_late_seconds or include_late_minutes or include_late_hours or include_late_days:
+            for time_unit, late_events in events_log["late_events"].items():
```

Comment:
this gets a bit nested, what about breaking it out into a function?

Reply:
agreed. I'll work on that.

Reply:
> Flat is better than nested

You already know this, but just in case as reference: https://www.youtube.com/watch?v=3dUKpvSBgLE
(I need to do some more refactoring videos, feel free to share ideas or needs)

Reply:
Awesome, love that (you should share that in Circle as a WIN!)

I am also happy to see its philosophy applied in other frameworks, e.g. https://www.linkedin.com/posts/bbelderbos_python-softwaredevelopment-designpatterns-activity-7152598153521750016-vK_O?utm_source=share&utm_medium=member_desktop

I need to finish my writings and maybe make an ebook, so far I have:
https://github.com/bbelderbos/bobcodesit/blob/main/index.md#zen

Reply:
great post! And the tips archive is incredible!